linux/devdoc/async_socket_linux_requirements.md

```diff
@@ -231,9 +191,7 @@ MOCKABLE_FUNCTION(, ASYNC_SOCKET_SEND_SYNC_RESULT, async_socket_send_async, ASYN
 
   - **SRS_ASYNC_SOCKET_LINUX_11_056: [** `async_socket_send_async` shall create a context for the send where the `payload`, `on_send_complete` and `on_send_complete_context` shall be stored. **]**
```

Comment:
>create [](http://example.com/codeflow?start=73&length=6)

so let's assume there are 3 `buffer_count`. Let's assume the 1 buffer will be send synchronously (great joy!).

Let's assume the second buffer gets a "EWOULDBLOCK" so it is queued for notifications.

Let's assume the 3 buffer fails in this "create" (any create can fail).

So this leave us with:

1 packet "on the wire"
1 payload queued for sending whenever the completion port so pleases to tell us
1 payload failed (because malloc fails for it).

As per 11_063 the function will return "ERROR".

What's the mitigation at user level? This is a very delicate problem which frankly I don't see how it can be tackled with the current function prototype.

Reply:
That's a reason to wait, because curiosity is eating me.

Reply:
Ok, this API is kinda confusing to me and I didn't want to change it because of windows API would have to change too, but the send_async is only called with 1 buffer.  I think we should change this from being an array to be a singular ASYNC_SOCKET_BUFFER.

Reply:
I will run this by our fearless leader and see if I should change it (this will be done in a different PR btw).